Added check if the device is a keyboard

cgnmause opened the pull request:
Hi,
I hope I got forking, branching and generating pull requests right now.
I would like to introduce my fix for the issue#231.
The KKMOON RFID Reader appears twice in the devices list as HID 413d:2107. One device is a mouse device and the other one is the keyboard device (we are interested in that).
Unfortunately the current reader code will pick the mouse device which cannot work.
My fix will additionally check if the device is a keyboard.

Any thoughts on my solution? 
Best regards
Christoph

p.s.: I just could test the code against the KKMOON reader. Can anyone check the code with other usb readers as well?

MiczFlor commented:
HI @cgnmause 
I added one question to your pull request. Otherwise it looks good. I will test it. But that might take some time - just started a new job :)

cgnmause commented:
 

Hi Micz,

I am sorry. I have not checked my mail for a couple of days. Surely you can remove one existance of ecodes. I must have overseen the exisrting one when I added it.

 

Best regards

Christoph

 


Gesendet: Samstag, 24. November 2018 um 23:24 Uhr
Von: "Micz Flor" <notifications@github.com>
An: MiczFlor/RPi-Jukebox-RFID <RPi-Jukebox-RFID@noreply.github.com>
Cc: cgnmause <c.mause@web.de>, Mention <mention@noreply.github.com>
Betreff: Re: [MiczFlor/RPi-Jukebox-RFID] Added check if the device is a keyboard (#327)


@MiczFlor commented on this pull request.

Hi @cgnmause
Thanks for you addition to the code. I added a question in the file a few days ago: https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/327/files
could you please take a look?
All the best, micz


In scripts/Reader.py:

> @@ -5,20 +5,31 @@
 import os.path
 import sys
 
-from evdev import InputDevice, categorize, ecodes, list_devices
+from evdev import InputDevice, categorize, ecodes, list_devices, ecodes


@cgnmause
does it matter that ecodes is twice in this list?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

cgnmause commented:
> HI @cgnmause
> I added one question to your pull request. Otherwise it looks good. I will test it. But that might take some time - just started a new job :)

Good luck on your new job! Thank you very much for that project. It helped me lot.

MiczFlor commented:
Hi @cgnmause 
I am wondering what's the best thing to do. Possibly add your solution as a special KKMOON script? And add good documentation in the wiki?
The reason I am hesitating with this request is because it solves one problem for one device for one person. And the last time I merged changes to the Reader.py, the bug reporting exploded, because the solution created more havoc than solve problems.
What - other than the keyboard issue - is solved with your pull request?

MiczFlor commented:
Hi @cgnmause 
merging this with the `develop` branch. Please check if this works, if you find the time.
Thanks for the contribution, I keep my fingers crossed that it does not crash on other Readers :)
